### PR TITLE
QSPIF Block Device

### PR DESCRIPTION
**Pull request by offirko**

### CODE REVIEW

**Comment by MarceloSalazar** [comment]
Thanks for sharing. Is looking good!

Just a reminder, we'll need to update this page as well, add info on QSPI and how to run tests:
https://os.mbed.com/docs/v5.9/reference/blockdevice-port.html

**Comment by AnotherButler** [comment]
ping @yossi2le

**Comment by yossi2le** [comment]
@AnotherButler 
Sorry about the delay, I will look into it tonight or tomorrow morning.

**Comment by AnotherButler** [comment]
@yossi2le Thanks so much. I typically wouldn't pink you less than 24 hours after tagging you (because that's kind of obnoxious), but I wanted to make sure you could get to this before the end of the week, so it can go into out-of-box testing.

**Comment by AnotherButler** [comment]
Note: Once the code merges, we'll need to update the class reference and example.

**Comment by AnotherButler** [comment]
I see the code for this isn't going onto master for 5.10, so I'll wait to merge the docs.

**Comment by AnotherButler** [comment]
Is this code going onto master for 5.11?

**Comment by MarceloSalazar** [comment]
The code for QSPI is already in master
https://github.com/ARMmbed/mbed-os/tree/master/components/storage/blockdevice/COMPONENT_QSPIF

File: docs/reference/api/storage/QSPIFBlockDevice.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+![QSPIFBlockDevice](QSPIFBlockDevice_Class1.jpg)<span> QSPIFBlockDevice class hierarchy</span></span>
+
+The QSPIFBlockDevice is a block device driver for NOR-based QSPI Flash devices that support the SFDP standard. NOR-based QSPI Flash supports up to 4 bits per cycle of instruction, address and data. SFDP-based QSPI Flash supports variable bus modes (single, dual and quad), several sector erase size types and multiple regions of sector size types.
+
+For more information about the SFDP JEDEC standard, please see [its documentation](https://www.jedec.org/system/files/docs/JESD216C.pdf).
+
+### QSPIFBlockDevice class reference
+
+[![View code](https://www.mbed.com/embed/?type=library)](http://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/class_heap_block_device.html)
```
Comment (AnotherButler) [inline docs/reference/api/storage/QSPIFBlockDevice.md]: This still has HeapBlockDevice as the placeholder. Can someone please help me find QSPIFBlockDevice in the Doxy: https://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/annotated.html? I haven't been able to find it.

File: docs/reference/api/storage/QSPIFBlockDevice.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+### QSPIFBlockDevice example
+
+This example creates a QSPIFBlockDevice, erases a sector, programs it, reads the block back and cleans up.
+
+[![View code](https://www.mbed.com/embed/?url=https://os.mbed.com/teams/mbed_example/code/QSPIFBlockDevice_ex_1/)](https://os.mbed.com/teams/mbed_example/code/QSPIFBlockDevice_ex_1/file/5991e7053465/main.cpp)
```
Comment (AnotherButler) [inline docs/reference/api/storage/QSPIFBlockDevice.md]: This link doesn't work for me. Where do I find the example?

Comment (AnotherButler) [reply docs/reference/api/storage/QSPIFBlockDevice.md]: ping @offirko

File: docs/reference/api/storage/QSPIFBlockDevice.md (hunk 1 above)
Comment (offirko) [reply docs/reference/api/storage/QSPIFBlockDevice.md]: @AnotherButler - sorry for the delay.
Could you possibly update the link to: 
https://os.mbed.com/teams/mbed_example/code/QSPIFBlockDevice_example/file/5608693b47aa/main.cpp/

File: docs/reference/api/storage/QSPIFBlockDevice.md (hunk 2 above)
Comment (offirko) [reply docs/reference/api/storage/QSPIFBlockDevice.md]: @AnotherButler - sorry for the delay.
Could you possibly update the link to:
https://os.mbed.com/teams/mbed_example/code/QSPIFBlockDevice_example/file/5608693b47aa/main.cpp/

Comment (offirko) [reply docs/reference/api/storage/QSPIFBlockDevice.md]: @AnotherButler - sorry for the delay.
I updated the link.

Comment (offirko) [reply docs/reference/api/storage/QSPIFBlockDevice.md]: @AnotherButler - would you be able to assist with the doxygen Class Reference link?

**Comment by AnotherButler** [comment]
Thanks for updating the example link. I've updated the class reference. Is this ready to merge?

**Comment by offirko** [comment]
Yes, its ready for merge. Thanks for your help!

**Pull request merged**